lib/resourcebuilder/batch.go

```diff
@@ -48,7 +48,7 @@ func checkJobHealth(ctx context.Context, client batchclientv1.JobsGetter, job *b
 	// Since we have filled in "activeDeadlineSeconds",
 	// the Job will 'Active == 0' if and only if it exceeds the deadline.
 	// Failed jobs will be recreated in the next run.
-	if j.Status.Active == 0 && j.Status.Failed > 0 {
+	if j.Status.Active == 0 && len(j.Status.Conditions) > 0 {
```

Comment:
```suggestion
	if j.Status.Active == 0{
```
Perhaps we should leave out conditions check and have it checked later?

Reply:
I messed up the implementation. I've fixed it now. PTAL.